core/tracing.go

```diff
@@ -29,7 +30,7 @@ func StartDutyTrace(ctx context.Context, duty Duty, spanName string, opts ...tra
 	ctx, outerSpan = tracer.Start(tracer.RootedCtx(ctx, traceID), fmt.Sprintf("core/duty.%s", strings.Title(duty.Type.String())))
 	ctx, innerSpan = tracer.Start(ctx, spanName, opts...)
 
-	outerSpan.SetAttributes(attribute.Int64("slot", duty.Slot))
+	outerSpan.SetAttributes(attribute.Int64("slot", safeInt64(duty.Slot)))
```

Comment:
Same here, i wonder if we can have Slot as uint64 and eventually do the safeint64 checks when creating the struct.

Reply:
attribute has only 3 methods: `attribute.Int()`, `attribute.Int64()` & `attribute.String()`

q: why would we need to perform safeint checks if we define slots are uint64, won't the compiler already catch this?